Minecraft.Client.Test/Fakes/FakeConnection.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Decent.Minecraft.Client.Test.Fakes
+{
+    public class FakeConnection : IConnection
+    {
+        readonly List<KeyValuePair<string, IList>> _commandHistory = new List<KeyValuePair<string, IList>>();
+
+        public KeyValuePair<string, IList> GetLastCommand()
+        {
+            return _commandHistory.Last();
+        }
+
+        public KeyValuePair<string, IList> GetLastCommand(string command)
+        {
+            return _commandHistory.Last(c => c.Key.EndsWith(command));
+        }
+
+        public void Dispose() { }
+        public void Close() { }
+        public void Open() { }
+        public Task OpenAsync() { return Task.CompletedTask; }
+
+        public string Address { get; set; }
+        public int Port { get; set; }
+
+
+        public Task<string> SendAndReceiveAsync(string function, params object[] data)
+        {
+            return Task.Run(() =>
+            {
+                var command = new KeyValuePair<string, IList>(function, data);
+                Console.WriteLine($"Sendng and receiving: {command.Key}");
+                _commandHistory.Add(command);
+
+                if (command.Key.EndsWith(".getPos"))
+                {
+                    if (_commandHistory.Any(c => c.Key.EndsWith(".setPos")))

```

Comment:
Yes, this was a very blunt implementation I admit, but I usually try to implement the simplest thing I can think of first, then refactor when it gets more complicated. Not sure how this will be used (if at all), but using a dictionary seems like the right way to go forward :)